modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java

```diff
@@ -402,7 +402,7 @@ private CompletableFuture<AsyncSqlCursor<List<Object>>> querySingle0(
 
                                 // Transactional DDL is not supported as well as RO transactions, hence
                                 // only DML requiring RW transaction is covered
-                                boolean implicitTxRequired = plan.type() == Type.DML && outerTx == null;
+                                boolean implicitTxRequired = (plan.type() == Type.DML || plan.type() == Type.QUERY) && outerTx == null;
```

Comment:
Could you please explain what are you doing here?
First of all, what's the point of using implicit transactions inside sql engine?
What is "QUERY" type? and why we should create implicit transaction if one?

Reply:
We should create a transaction into SQL for all DML operation and QUERY (it is SELECT operation) too, because all the operations are transactional.

Reply:
Make sense, thanks.